components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -263,7 +267,7 @@ public Action updateActionEndpointAuthProperties(String actionId, AuthType authe
         Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
         updateActionEndpointAuthProperties(dbConnection, actionId, authentication, tenantId);
         IdentityDatabaseUtil.closeConnection(dbConnection);
-        return getActionByActionId(actionId, tenantId);
+        return getActionByActionId(authentication.getType().getType(), actionId, tenantId);
```

Comment:
Is this right ? This says authencation.getType

Reply:
This is fixed now.